zebra-chain/src/parameters/network/subsidy.rs

```diff
@@ -60,76 +88,144 @@ pub const FUNDING_STREAM_RECEIVER_DENOMINATOR: u64 = 100;
-        let mut hash_map = HashMap::new();
-        hash_map.insert(FundingStreamReceiver::Ecc, 7);
-        hash_map.insert(FundingStreamReceiver::ZcashFoundation, 5);
-        hash_map.insert(FundingStreamReceiver::MajorGrants, 8);
-        hash_map
+    numerator: u64,
+    /// Addresses for the funding stream recipient
+    addresses: Vec<transparent::Address>,
+}
+
+impl FundingStreamRecipient {
+    /// Creates a new [`FundingStreamRecipient`].
+    pub fn new<I, T>(numerator: u64, addresses: I) -> Self
+    where
+        T: ToString,
+        I: IntoIterator<Item = T>,
+    {
+        Self {
+            numerator,
+            addresses: addresses
+                .into_iter()
+                .map(|addr| {
+                    let addr = addr.to_string();
+                    addr.parse()
+                        .expect("funding stream address must deserialize")
+                })
+                .collect(),
+        }
+    }
+
+    /// Returns the numerator for this funding stream.
+    pub fn numerator(&self) -> u64 {
+        self.numerator
+    }
+
+    /// Returns the receiver of this funding stream.
+    pub fn addresses(&self) -> &[transparent::Address] {
+        &self.addresses
+    }
+}
+
+lazy_static! {
+    /// The pre-NU6 funding streams for Mainnet
+    pub static ref PRE_NU6_FUNDING_STREAMS_MAINNET: FundingStreams = FundingStreams {
+        height_range: Height(1_046_400)..Height(2_726_400),
+        recipients: [
+            (
+                FundingStreamReceiver::Ecc,
+                FundingStreamRecipient::new(7, FUNDING_STREAM_ECC_ADDRESSES_MAINNET.iter()),
+            ),
+            (
+                FundingStreamReceiver::ZcashFoundation,
+                FundingStreamRecipient::new(5, FUNDING_STREAM_ZF_ADDRESSES_MAINNET),
+            ),
+            (
+                FundingStreamReceiver::MajorGrants,
+                FundingStreamRecipient::new(8, FUNDING_STREAM_MG_ADDRESSES_MAINNET),
+            ),
+        ]
+        .into_iter()
+        .collect(),
     };
 
-    /// Start and end Heights for funding streams
-    /// as described in [protocol specification §7.10.1][7.10.1].
-    ///
-    /// [7.10.1]: https://zips.z.cash/protocol/protocol.pdf#zip214fundingstreams
-    // TODO: Move the value here to a field on `testnet::Parameters` (#8367)
-    pub static ref FUNDING_STREAM_HEIGHT_RANGES: HashMap<NetworkKind, std::ops::Range<Height>> = {
-        let mut hash_map = HashMap::new();
-        hash_map.insert(NetworkKind::Mainnet, Height(1_046_400)..Height(2_726_400));
-        hash_map.insert(NetworkKind::Testnet, Height(1_028_500)..Height(2_796_000));
-        hash_map.insert(NetworkKind::Regtest, Height(1_028_500)..Height(2_796_000));
-        hash_map
+    /// The post-NU6 funding streams for Mainnet
+    pub static ref POST_NU6_FUNDING_STREAMS_MAINNET: FundingStreams = FundingStreams {
+        height_range: Height(2_726_400)..Height(3_146_400),
```

Comment:
The start range is currently defined in the ZIP (https://github.com/zcash/zips/pull/878/files) but the end range is not. Where is this number coming from ? Should we clarify ?

Reply:
Another thing that had my attention is that there might be no funding stream at all after NU6. Should we still want to have a `POST_NU6_FUNDING_STREAMS_MAINNET` around if that is the case?

Reply:
> The start range is currently defined in the ZIP (https://github.com/zcash/zips/pull/878/files) but the end range is not. Where is this number coming from ? Should we clarify ?

These heights and the recipients in the next PR are based on alternative 2 in the lockbox funding allocation draft ZIP, updated the docs to explain where it's from in https://github.com/ZcashFoundation/zebra/pull/8718/commits/852d63438288cd433e6c2cbd88c4326ba4765e56.